modules/ROOT/generate-docs.sh

```diff
@@ -2,14 +2,42 @@
 
 set -exuo pipefail
 
-cd $(dirname $0)
-
 GIT_ROOT=$(git rev-parse --show-toplevel)
 
-PATH_TO_SYNC_GATEWAY=${PATH_TO_SYNC_GATEWAY:-${GIT_ROOT}/../sync_gateway/}
+# Get upstream sync_gateway repo
+cd "${GIT_ROOT}"
+branch=release/3.2.1
+
+[[ -d ./sync_gateway ]] || git clone --no-checkout https://github.com/couchbase/sync_gateway.git
+cd sync_gateway
+git sparse-checkout init --cone
+git sparse-checkout set docs/api
```

Comment:
I'd be cautious about pulling a subset of directories and this repo is pretty small. This needs to have .redocly.yaml - https://github.com/couchbase/sync_gateway/blob/main/.redocly.yaml which does some processing of documentation using javascript.

Reply:
That's okay, `sparse-checkout` gets all the files in each directory in the path, so `.redocly.yaml` is included in the  checkout